### ISSUE TITLE
Add sub-directory property to Git fetcher

### ISSUE THREAD

**Issue by trevor-vaughan** [inspec/inspec#1974]
### Description

In my experience, various users have issues with large numbers of closely related repositories being combined into a reference platform.

See https://github.com/chris-rock/acme-inspec-profile/blob/master/inspec.yml#L10-L21 for a small example.

It would be nice to be able to specify named subdirectories for those projects that wish to use a single repository for InSpec content.

Example:

```yaml
depends:
  - name: my_inspec
    url: https://github.com/me/my_inspec_bundle
    directory: special_sauce
```

This would point `my_inspec` at the `special_sauce` subdirectory of `my_inspec_bundle` and improve the user experience.

**Comment by chris-rock** [comment]
Once we harmonized the vendoring mechanism https://github.com/chef/inspec/issues/1659, one solution could be to set the path:
```
depends:
  - name: my_inspec
    path: vendor/my_inspec_bundle
```

**Comment by chris-rock** [comment]
@trevor-vaughan would that approach work for you?

**Comment by trevor-vaughan** [comment]
@chris-rock So, I definitely need the ability to pull from Git directly since I can readily have that everywhere with no changes to infrastructure. The `path` key would work fine as long as it just starts at the top level of the repo/unpacked directory/whatever and looks down from there.

**Comment by stevemac007** [comment]
Any thoughts on this feature?

We have profiles embedded with our provisioning code in sub-directories and would love to compose those into other repositories as mentioned.

Needing a separate git repo for the tests breaks the consistency of the git repos, requiring commits across repositories.

**Comment by naunga** [comment]
Looks like this is a dead'ish issue, but we have the exact same needs and philosophy as @stevemac007. 

We put our InSpec profiles in the same repos as the Chef recipes they apply to.

**Comment by jerryaldrichiii** [comment]
@naunga glad you quantified it with `ish` :wink: 

@stevemac007 @naunga @trevor-vaughan after #3286 merges and is released you should be able use `path://` and it will vendor it into `vendor/`.

**Comment by jerryaldrichiii** [comment]
@trevor-vaughan @naunga @stevemac007 I may have misread this initially. With my changes in #3286 you will be able to specify a local path and have it vendored when running `inpsec vendor`, is that sufficient?

NOTE: It would not clone/download a repo for you and then let you specify a path on that repo.

**Comment by clintoncwolfe** [comment]
From the last sentence of the OP, I think this is a request to add the ability to fetch a profile by git, then have inspec look in a subdirectory of the cloned repo.  I don't follow @chris-rock's statement that vendor harmonization would address this; I think it's simply a separate feature request.

**Comment by jerryaldrichiii** [comment]
I agree, I'll flag it as such as it's a valid request.

**Comment by trevor-vaughan** [comment]
Sorry that I'm late to the party. This is a separate feature that I need for composing profiles from multiple sources that have them under the root level of the repo.

Specifically, this allows for micro-embedding profiles with components and creating a higher level one later which would allow software projects to embed inspec checks for their own software as they see fit.

**Comment by kekaichinose** [comment]
Closing this issue as it is being addressed in #4217
